micro/reify_test.go

```diff
@@ -28,44 +27,44 @@ scheme code:
 	)
 */
 func TestReify(t *testing.T) {
-	a1 := "(,x . (,u ,w ,y ,z ((ice) ,z)))"
-	a2 := "(,y . corn)"
-	a3 := "(,w . (,v ,u))"
-	s := "(" + a1 + " " + a2 + " " + a3 + ")"
-	e, err := sexpr.Parse(s)
-	if err != nil {
-		t.Fatal(err)
-	}
+	u := ast.NewVar("u", 0)
+	v := ast.NewVar("v", 1)
+	w := ast.NewVar("w", 2)
+	x := ast.NewVar("x", 3)
+	y := ast.NewVar("y", 4)
+	z := ast.NewVar("z", 5)
+	a1 := ast.NewList(x, u, w, y, z, ast.NewList(ast.NewList(ast.NewSymbol("ice")), z))
+	a2 := ast.NewList(y, ast.NewSymbol("corn"))
+	a3 := ast.NewList(w, v, u)
+	e := ast.NewList(a1, a2, a3)
 	ss := Substitutions{
-		"x": e.Car().Cdr(),
-		"y": e.Cdr().Car().Cdr(),
-		"w": e.Cdr().Cdr().Car().Cdr(),
-	}
-	if !e.IsPair() {
-		t.Fatalf("expected list")
+		3: e.Car().Cdr(),
+		4: e.Cdr().Car().Cdr(),
+		2: e.Cdr().Cdr().Car().Cdr(),
 	}
-	gote := ReifyVarFromState("x")(&State{Substitutions: ss})
+	gote := ReifyIntVarFromState(3)(&State{Substitutions: ss})
 	got := gote.String()
-	want := "(_0 (_1 _0) corn _2 ((ice) _2))"
+	want := "(_0 (_1 _0) (corn) _2 ((ice) _2))"
```

Comment:
Yes, again here, I think was because of the improper list before.  It should be fixable.